src/events/endpoints.go

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright © 2020, Oracle and/or its affiliates. All rights reserved.
+ * The Universal Permissive License (UPL), Version 1.0
+ */
+package events
+
+import (
+	"github.com/go-kit/kit/endpoint"
+	"github.com/go-kit/kit/tracing/opentracing"
+	stdopentracing "github.com/opentracing/opentracing-go"
+	"golang.org/x/net/context"
+)
+
+// Endpoints collects the endpoints that comprise the Service.
+type Endpoints struct {
+	EventsEndpoint endpoint.Endpoint
+	HealthEndpoint endpoint.Endpoint
+}
+
+// MakeEndpoints returns an Endpoints structure, where each endpoint is
+// backed by the given service.
+func MakeEndpoints(s Service, tracer stdopentracing.Tracer) Endpoints {
+	return Endpoints{
+		EventsEndpoint: opentracing.TraceServer(tracer, "POST /events")(MakeEventsEndpoint(s)),
+		HealthEndpoint: opentracing.TraceServer(tracer, "GET /health")(MakeHealthEndpoint(s)),
+	}
+}
+
+// MakeEventsEndpoint returns an endpoint via the given service.
+func MakeEventsEndpoint(s Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		var span stdopentracing.Span
+		span, ctx = stdopentracing.StartSpanFromContext(ctx, "receive events")
+		span.SetTag("service", "events")
+		defer span.Finish()
+		req := request.(EventsRequest)
+		received, err := s.EventsReceiver(req.Source, req.Track, req.Events)
+		return EventsResponse{EventsReceived: received, Err: err}, nil
+	}
+}
+
+// MakeHealthEndpoint returns current health of the given service.
+func MakeHealthEndpoint(s Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+		var span stdopentracing.Span
+		span, ctx = stdopentracing.StartSpanFromContext(ctx, "health check")
+		span.SetTag("service", "events")
+		defer span.Finish()
+		health := s.Health()
+		return healthResponse{Health: health}, nil
+	}
+}
+
+// EventsRequest represents a request for events tracking
+type EventsRequest struct {
```

Comment:
nit: similar as above -- the package is named `events`, so having the name `Events` repeat in structs is odd. (e.g. `events.EventsResponse` or `events.EventsRequest`). It should be `events.Response` and `events.Request`

Reply:
I think this should be the left as-is, since there is more than one `Request` type in the service.